docs/tutorial/code/fastapi/task.yaml

```diff
@@ -0,0 +1,137 @@
+  curl localhost:8000
+  # [docs:curl-fastapi-end]
+
+  kill $!
+
+  # [docs:create-rockcraft-yaml]
+  rockcraft init --profile fastapi-framework
+  # [docs:create-rockcraft-yaml-end]
+  sed -i "s/name: .*/name: fastapi-hello-world/g" rockcraft.yaml
+
+  # [docs:pack]
+  ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=true rockcraft pack
+  # [docs:pack-end]
+
+  # [docs:ls-rock]
+  ls *.rock -l --block-size=MB
+  # [docs:ls-rock-end]
+
+  # [docs:skopeo-copy]
+  sudo rockcraft.skopeo --insecure-policy \
+    copy oci-archive:fastapi-hello-world_0.1_amd64.rock \
+    docker-daemon:fastapi-hello-world:0.1
+  # [docs:skopeo-copy-end]
+
+  # [docs:docker-images]
+  sudo docker images fastapi-hello-world:0.1
+  # [docs:docker-images-end]
+
+  # [docs:docker-run]
+  sudo docker run --rm -d -p 8000:8000 \
+    --name fastapi-hello-world fastapi-hello-world:0.1
+  # [docs:docker-run-end]
+  retry -n 5 --wait 2 curl localhost:8000
+
+  # [docs:curl-fastapi-rock]
+  curl localhost:8000
+  # [docs:curl-fastapi-rock-end]
+
+  # [docs:get-logs]
+  sudo docker exec fastapi-hello-world pebble logs fastapi
+  # [docs:get-logs-end]
+
+  # [docs:stop-docker]
+  sudo docker stop fastapi-hello-world
+  sudo docker rmi fastapi-hello-world:0.1
+  # [docs:stop-docker-end]
+
+  # [docs:change-base]
+  sed -i \
+    "s/base: .*/base: bare\nbuild-base: ubuntu@24.04/g" \
+    rockcraft.yaml
+  # [docs:change-base-end]
+  sed -i "s/version: .*/version: 0.1-chiselled/g" rockcraft.yaml
+
+  # [docs:chisel-pack]
+  ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=true rockcraft pack
+  # [docs:chisel-pack-end]
+
+  # [docs:ls-bare-rock]
+  ls *.rock -l --block-size=MB
+  # [docs:ls-bare-rock-end]
+
+  # [docs:docker-run-chisel]
+  sudo rockcraft.skopeo --insecure-policy \
+    copy oci-archive:fastapi-hello-world_0.1-chiselled_amd64.rock \
+    docker-daemon:fastapi-hello-world:0.1-chiselled
+  sudo docker images fastapi-hello-world:0.1-chiselled
+  sudo docker run --rm -d -p 8000:8000 \
+    --name fastapi-hello-world fastapi-hello-world:0.1-chiselled
+  # [docs:docker-run-chisel-end]
+  retry -n 5 --wait 2 curl localhost:8000
+
+  # [docs:curl-fastapi-bare-rock]
+  curl localhost:8000
+  # [docs:curl-fastapi-bare-rock-end]
+
+  # [docs:stop-docker-chisel]
+  sudo docker stop fastapi-hello-world
+  sudo docker rmi fastapi-hello-world:0.1-chiselled
+  # [docs:stop-docker-chisel-end]
+
+  cat time_app.py > app.py
+  sed -i "s/version: .*/version: 0.2/g" rockcraft.yaml
+
+  # [docs:docker-run-update]
+  ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=true rockcraft pack
+  sudo rockcraft.skopeo --insecure-policy \
+    copy oci-archive:fastapi-hello-world_0.2_amd64.rock \
+    docker-daemon:fastapi-hello-world:0.2
+  sudo docker images fastapi-hello-world:0.2
+  sudo docker run --rm -d -p 8000:8000 \
+    --name fastapi-hello-world fastapi-hello-world:0.2
+  # [docs:docker-run-update-end]
+  retry -n 5 --wait 2 curl localhost:8000/time
+
+  # [docs:curl-time]
+  curl localhost:8000/time
```

Comment:
I think by default curl does not fail if there is a 404 error. Maybe it could be useful to add a `--fail` argument (or another curl after the [docs] to check for the desired response

Reply:
Good point, added for all the existing tutorials as well